app/helpers/shop_helper.rb

```diff
@@ -20,27 +20,23 @@ def user_is_related_to_distributor?
     )
   end
 
-  def base_shop_tabs(column_sizes)
+  def shop_tabs
     [
-      { name: 'about', cols: column_sizes[0],
-        title: t(:shopping_tabs_about, distributor: current_distributor.name) },
-      { name: 'producers', cols: column_sizes[1],
-        title: t(:label_producers) },
-      { name: 'contact', cols: column_sizes[2],
-        title: t(:shopping_tabs_contact) }
-    ]
-  end
-
-  def tabs_with_groups
-    tabs = base_shop_tabs([6, 2, 2])
-    tabs << { name: 'groups', title: t(:label_groups), cols: 2 }
+      { name: 'home', title: t(:shopping_tabs_home), show: show_home_tab? },
+      { name: 'shop', title: t(:shopping_tabs_shop), show: !require_customer? },
+      { name: 'about', title: t(:shopping_tabs_about), show: true },
+      { name: 'producers', title: t(:label_producers), show: true },
+      { name: 'contact', title: t(:shopping_tabs_contact), show: true },
+      { name: 'groups', title: t(:label_groups), show: current_distributor.groups.any? },
+    ].select{ |tab| tab[:show] }
   end
 
-  def tabs_without_groups
-    base_shop_tabs([4, 4, 4])
+  def show_home_tab?
+    require_customer? || current_distributor.preferred_shopfront_message.present?
   end
 
-  def shop_tabs
-    current_distributor.groups.present? ? tabs_with_groups : tabs_without_groups
+  def shopfront_closed_message?
+    @order_cycles && @order_cycles.empty? &&
+      current_distributor.preferred_shopfront_closed_message.present?
```

Comment:
can you extract "@order_cycles && @order_cycles.empty?" to a method a give it a good name. does it mean  "no_open_order_cycles?"